.github/workflows/release-docker.yml

```diff
@@ -0,0 +1,56 @@
+name: Release to Docker Hub
+
+on:
+  release:
+    types: [published]
+
+env:
+  # Only used locally for validating image
+  TEST_TAG: "${{ env.DOCKER_IMAGE_NAME }}:test"
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      # Determine the tags to publish based on the release tag
+      - name: Docker Metadata
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            ${{ env.DOCKER_IMAGE_NAME }}
+          tags: |
+            # publish by version on all release tags, and `latest` on stable release tags.
+            type=semver,pattern={{version}},latest=${{ !github.event.release.prerelease }}}
+
+            # Originally, I thought that `type=semver,value=rc` would work,
+            # but the `value` is actually used to replace the input value,
+            # not the output value, which is not clear from the documentation.
+            type=match,pattern=v(\d+.\d+.\d+-(rc).\d+),group=2,latest=false
+
+      # Build and test the image
+      - name: Build and Test
+        uses: docker/build-push-action@v4
+        with:
+          load: true
+          tags: ${{ env.TEST_TAG }}
+          build-args: |
+            pyquil_version=${{ github.event.release.tag_name }}
```

Comment:
This requires that [`workflows/release.yml`](https://github.com/rigetti/pyquil/blob/v4/.github/workflows/release.yml) has finished publishing to pypi.

There are two approaches to this:
1. Merge this file with the `release.yml` file - this is the easiest, but the downside is that it will have to either have to skip `workflow_dispatch`, or handle it explicitly.
2. Use some sort of chain-calling hack like https://www.r-bloggers.com/2020/07/running-github-actions-sequentially/, which could be fairly messy as I think we would lose access to the `gihub.event.release` event